urle381b6cba7_IkarosKappler_ikrs_http_tgz-pJ8-ikrs_json_rpc_JSONRPCHandlerJ8/src/ikrs/httpd/ErrorResponseBuilder.java

```diff
@@ -0,0 +1,429 @@
+     *                    map contents depends on the underlying implementation; some
+     *                    builders even allow null-additionals.
+     *
+     * @return A new HTTPRequest built from the HTTP headers.
+     *
+     **/
+    public PreparedHTTPResponse create( HTTPHeaders headers,
+					PostDataWrapper postData,
+					UUID socketID,
+					Socket socket,
+					UUID sessionID, 
+					Map<String,BasicType> additionals
+					) {
+
+	this.getHTTPHandler().getLogger().log( Level.INFO,
+					       getClass().getName(),
+					       "socketID=" + socketID + ", sessionID=" + sessionID + ", additionalSettings=" + additionals 
+					       );
+	
+
+	BasicType statusCode   = additionals.get( "statusCode" );
+	BasicType reasonPhrase = additionals.get( "reasonPhrase" );
+	BasicType errorMessage = additionals.get( "errorMessage" );
+
+	try {
+	    
+
+	    if( statusCode == null || reasonPhrase == null || errorMessage == null ) {
+
+		this.getHTTPHandler().getLogger().log( Level.WARNING,
+						       getClass().getName(),
+						       "Cannot build error reply. The passed additionals miss some essential params (statusCode="+statusCode+", reasonPhrase="+reasonPhrase+", errorMessage="+errorMessage+"). Senging INTERNAL_SERVER_ERROR. additionals=" + additionals.toString() 
+						       );
+
+	
+		return buildPreparedErrorResponse( headers, 
+						   socketID, 
+						   socket, 
+						   sessionID, // userID,
+						   
+						   Constants.HTTP_STATUS_SERVERERROR_INTERNAL_SERVER_ERROR,
+						   "Internal Server Error",
+						   "The system encountered a fatal error. Your request could not be processed.",
+
+						   additionals
+						   );
+	    } else {
+
+		return buildPreparedErrorResponse( headers, 
+						   socketID, 
+						   socket, 
+						   sessionID,   // userID,
+
+						   statusCode.getInt(),
+						   reasonPhrase.getString(),
+						   errorMessage.getString(),
+
+						   additionals
+						   );
+
+	    }
+
+	} catch( BasicTypeException e ) {
+
+	    this.getHTTPHandler().getLogger().log( Level.WARNING,
+						       getClass().getName(),
+						       "Cannot build error reply. The passed additionals miss some essential params (statusCode="+statusCode+", reasonPhrase="+reasonPhrase+", errorMessage="+errorMessage+"). Senging INTERNAL_SERVER_ERROR." 
+						       );
+		
+	    return buildPreparedErrorResponse( headers, 
+					       socketID, 
+					       socket, 
+					       sessionID, // userID,
+					       
+					       Constants.HTTP_STATUS_SERVERERROR_INTERNAL_SERVER_ERROR,
+					       "Internal Server Error",
+					       "The system encountered a fatal error. Your request could not be processed.",
+
+					       additionals
+					       );
+
+	}
+	
+
+    }
+    //---END-------------------------------- ReplyBuilder implementaion -----------------------
+
+
+    protected PreparedHTTPResponse buildPreparedErrorResponse( HTTPHeaders headers, 
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 22. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.